components/server/ee/src/prebuilds/prebuild-manager.ts

```diff
@@ -228,4 +238,33 @@ export class PrebuildManager {
         }
         return hostContext.contextParser;
     }
-}
\ No newline at end of file
+
+    private async shouldRateLimitPrebuild(span: opentracing.Span, cloneURL: string): Promise<boolean> {
+        const minuteAgo = secondsBefore(new Date().toISOString(), 60)
+        const unabortedCount = await this.workspaceDB.trace({span}).countUnabortedPrebuildsSince(cloneURL, new Date(minuteAgo))
+        const limit = this.getPrebuildRateLimitForCloneURL(cloneURL)
```

Comment:
nit:  we don't have a formatter yet, but if we were to have one we'd required `;`to be used :upside_down_face:

Reply:
Will fix. Also cutting a ticket for a formatter. I need a smart computer to do the boring stuff for me.

Reply:
Already exists here: https://github.com/gitpod-io/gitpod/issues/8066

Reply:
Thanks. Created https://github.com/gitpod-io/gitpod/issues/8654